prefect_snowflake/credentials.py

```diff
@@ -76,3 +92,30 @@ def _validate_auth_kwargs(cls, values):
                 f"One of the authentication keys must be provided: {auth_str}\n"
             )
         return values
+
+    @root_validator(pre=True)
+    def _validate_token_kwargs(cls, values):
+        """
+        Ensure an authorization value has been provided by the user.
+        """
+        authenticator = values.get("authenticator")
+        token = values.get("token")
+        if authenticator == "oauth" and not token:
+            raise ValueError(
+                "If authenticator is set to `oauth`, `token` must be provided\n"
```

Comment:
I don't think we need this trailing newline?
```suggestion
                "If authenticator is set to `oauth`, `token` must be provided"
```